projects/igniteui-angular/src/lib/grids/common/events.ts

```diff
@@ -121,21 +128,30 @@ export interface IPinColumnEventArgs extends IBaseEventArgs {
 export interface IPinColumnCancellableEventArgs extends IPinColumnEventArgs, CancelableEventArgs {
 }
 
-/** 
- * Represents event arguments related to events, that can occur for rows in a grid 
+/**
+ * Represents event arguments related to events, that can occur for rows in a grid
  * Example for events: adding, deleting, selection, transaction, etc.
  */
 export interface IRowDataEventArgs extends IBaseEventArgs {
+    /**
+     * @deprecated since version 17.0.0
+     * Use`rowData` instead
+     */
     data: any;
-    /** 
-     * Represents the unique key, the row can be associated with. 
+    rowData: any
+    /**
+     * @deprecated since version 17.0.0
+     * Represents the unique key, the row can be associated with.
      * Available if `primaryKey` exists
+     * use rowKey instead
      */
     primaryKey: any;
+    rowKey: any;
```

Comment:
Was the comment wrong before? Because this says it was only available with primaryKey set, as in this would've been named correctly if true

Reply:
Check if the rowPin event actually emits only primaryKey values